src/test/java/com/exasol/adapter/json/SqlDataTypeJsonSerializerTest.java

```diff
@@ -0,0 +1,104 @@
+package com.exasol.adapter.json;
+
+import com.exasol.adapter.metadata.DataType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.json.JsonObject;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@RunWith(JUnitPlatform.class)
+class SqlDataTypeJsonSerializerTest {
+    @Mock DataType dataType;
+
+    @Test
+    void testSerializeUnsupportedThrowsException() {
+        when(this.dataType.getExaDataType()).thenReturn(DataType.ExaDataType.UNSUPPORTED);
+        assertThrows(IllegalArgumentException.class,
+              () -> SqlDataTypeJsonSerializer.serialize(this.dataType));
+    }
+
+    @Test
+    void testSerializeDecimal() {
+        when(this.dataType.getExaDataType()).thenReturn(DataType.ExaDataType.DECIMAL);
+        when(this.dataType.getPrecision()).thenReturn(5);
+        when(this.dataType.getScale()).thenReturn(3);
+        final JsonObject jsonObject = SqlDataTypeJsonSerializer.serialize(this.dataType).build();
+        assertThat(jsonObject.getInt("precision"), equalTo(5));
```

Comment:
Please use `assertAll` around the two sub tests.

Reply:
done